plugins/languages/markdown/markview.nix

```diff
@@ -0,0 +1,93 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib.nixvim) defaultNullOpts;
+  inherit (lib) types;
+in
+lib.nixvim.neovim-plugin.mkNeovimPlugin config {
+  name = "markview";
+  originalName = "markview.nvim";
+  defaultPackage = pkgs.vimPlugins.markview-nvim;
+
+  maintainers = [ lib.maintainers.khaneliman ];
+
+  description = ''
+    An experimental markdown previewer for Neovim.
+
+    Supports a vast amount of rendering customization.
+    Please refer to the plugin's [documentation](https://github.com/OXY2DEV/markview.nvim/wiki/Configuration-options) for more details.
+  '';
+
+  settingsOptions = {
+    buf_ignore = defaultNullOpts.mkListOf types.str [ "nofile" ] "Buftypes to disable markview-nvim.";
+
+    mode = defaultNullOpts.mkListOf types.str [
+      "n"
+      "no"
+    ] "Modes where preview is enabled.";
+
+    hybrid_modes = defaultNullOpts.mkListOf types.str null "Modes where hybrid mode is enabled.";
+
+    callback = {
+      on_enable = defaultNullOpts.mkLuaFn' {
+        pluginDefault.__raw = ''
```

Comment:
mkLuaFn is a bit magic in that it transforms string definitions into rawLua config.

This is why I think we should deprecate it in new code. But because of the magic, we can't remove it from old code without breaking users' configs.

Anyway, if we're going to continue using it here, we don't need `__raw` in the `pluginDefault`.

Reply:
I think I used this approach in the last few. I'll update them to not do this either after this one.